onedal/covariance/incremental_covariance.py

```diff
@@ -58,8 +58,8 @@ def __init__(self, method="dense", bias=False, assume_centered=False):
         self._reset()
 
     def _reset(self):
-        self._partial_result = self._get_backend(
-            "covariance", None, "partial_compute_result"
+        self._partial_result = BaseEstimator._get_backend(
+            self, "covariance", None, "partial_compute_result"
```

Comment:
BaseEmpiricalCovariance is inherited from BaseEstimator. The trick here is that we inherit from BaseEstimatorSPMD which also has _get_backend. Thus, if we put self._get_backend here then SPMD backend would be called which does not contain partial_compute_result.

Reply:
I'd prefer a definition of _reset in the SPMD interface which uses a super call for locality code. That would be simpler for maintainers in the future to see why certain things are done.  At a minimum it needs a comment in the code.

Reply:
It is also necessary to redefine `partial_fit` because `partial_compute` also does not exist in the SPMD backend. I'm not sure if the code duplication is better idea than the currently implemented one

Reply:
and also have to add that super call would not work there because BaseEstimator and BaseEstimatorSPMD have the same methods and if the class has both of them as parents we anyway need to specify directly which of them should be called.

Reply:
I would ask you do refactoring after this PR merged. Currently this looks like a workaround.
Create some `BaseIncreamenatlEstimator`, where get_backend method depending on provided method name returns required backend. This will be common for all incremental algos.

Reply:
> Create some `BaseIncreamenatlEstimator`, where get_backend method depending on provided method name returns required backend. This will be common for all incremental algos.

this might be a good idea, I'll think about it